cluster-autoscaler/config/dynamic/node_group_spec.go

```diff
@@ -32,6 +34,36 @@ type NodeGroupSpec struct {
 	MaxSize int `json:"maxSize"`
 	// Specifies whether this node group can scale to zero nodes.
 	SupportScaleToZero bool
+	// specifies the agentpool type, used for vms pool only
```

Comment:
Agentpool and vm pool is an AKS specific concept.  config/dynamic falls outside of our cloudprovider, so when adding changes we should consider how the behavior affects others.

Reply:
Oh that makes sense! okay I will use a different approach them.

<Update>: change to use azure cache to track vms pools